frog/imports/api/sessions.js

```diff
@@ -129,16 +132,29 @@ Meteor.methods({
       });
     });
     setTeacherSession(sessionId);
+    return sessionId;
   },
   'flush.session': sessionId => {
-    const session = Sessions.findOne(sessionId)
-    const graphId = session.graphId
-
+    const session = Sessions.findOne(sessionId);
+    if (!session) {
+      return;
+    }
+    const graphId = session.graphId;
+    Sessions.remove(sessionId);
     Graphs.remove(graphId);
     Activities.remove({ graphId });
     Operators.remove({ graphId });
     Connections.remove({ graphId });
-
-    Sessions.remove(sessionId);
+  },
+  'sessions.restart': session => {
+    const graphId = session.fromGraphId;
+    if (!graphId || !session) {
+      return;
+    }
+    Meteor.call('flush.session', session._id);
+    const newSessionId = Meteor.call('add.session', graphId);
+    Meteor.call('session.joinall', newSessionId);
+    runSession(newSessionId);
+    nextActivity(newSessionId);
```

Comment:
I agree, I suggest we leave it in for now, it's useful for development (although it would be great to move towards more automated tests etc, but we'll still need it). In the future, we can replace it with something more robust, or remove it altogether.